pimd/pim_zebra.c

```diff
@@ -320,10 +320,8 @@ void pim_zebra_upstream_rpf_changed(struct pim_instance *pim,
 		 * We have detected a case where we might need
 		 * to rescan the inherited o_list so do it.
 		 */
-		if (up->channel_oil->oil_inherited_rescan) {
```

Comment:
The original intent of this check was two fold:

a) This was *stupid* expensive and we needed to only rescan when it was necessary.  Are you ruining performance for an edge case?
b) We hoped to better detect when to rescan and do it then but were unsure about the root cause of the problem so this measure was put into place.  

Are you sure you are not ruining performance and is there not a check we can do that would limit the rescan to the case listed in the commit message?